### ISSUE TITLE
How to use ViewType.AnyView.child() method?

### ISSUE THREAD

**Issue by taihojwn** [nalexn/ViewInspector#353]
With Xcode 16, I have fixed most of our tests by using `find` method but sometimes I still need to deal with `AnyView` and want to unwrap it. I've found the `public static func child(_ content: Content) throws -> Content` on `ViewType.AnyView` and I wonder how to use it? I don't see a method for grabbing `content` from an `Inspectable`.

Thank you

**Comment by nalexn** [comment]
Hey, there is a way to opt out of implicit `AnyView`s that Xcode inserts, [details here](https://github.com/nalexn/ViewInspector/issues/327#issuecomment-2452948216).

I'm not entirely sure what you mean by `sometimes I still need to deal with AnyView and want to unwrap it.`, if you could share a code example and explain what exactly you want to achieve, that'd help. The `child(_ content: Content)` is intentionally internal, you're not supposed to use that.

I would recommend you check out the tests of the library, here are the [tests for AnyView](https://github.com/nalexn/ViewInspector/blob/0.10.1/Tests/ViewInspectorTests/SwiftUI/AnyViewTests.swift)

**Comment by taihojwn** [comment]
Thanks for getting back to me. We don't really want to really fight the system and opt out of that change just for unit tests to pass.
> The child(_ content: Content) is intentionally internal, you're not supposed to use that

Yeah I figured so but it's marked as `public` today so I figured I should ask.

In our case, we have a ViewModifier that employed another ViewModifer, something like this

```swift
 public struct IntrinsicHeightModifier: ViewModifier {
    static let sizeViewID = "sizeView""
    let callback: (CGFloat) -> Void

    public func body(content: Content) -> some View {
        content
            .background {
                content
                    .fixedSize(horizontal: false, vertical: true)
                    .hidden()
                    .readHeight(callback) // used another view modifier here
                    .accessibilityIdentifier(IntrinsicHeightModifier.sizeViewID)
            }
    }
}
```

And in tests
```swift
var modifiedContent: InspectableView<ViewType.ViewModifierContent>!
var height: CGFloat!

let testView = Text("Hello World!")
                        .modifier(
                            IntrinsicHeightModifier(heightCallback: { h in
                                    height = h
                                }
                            )
                        )
let modifier = try testView.inspect().find(ViewType.Text.self).modifier(ReadIntrinsicHeightModifier.self)
modifiedContent = try? modifier.find(ViewType.ViewModifierContent.self)

let sizeView = try? modifiedContent?.background().find { try $0.accessibilityIdentifier() == IntrinsicHeightModifier.sizeViewID }.anyView()  // this return an `InspectableView<ViewType.AnyView>`

let sizeView2 = try? modifiedContent?.background().find { try $0.accessibilityIdentifier() == IntrinsicHeightModifier.sizeViewID }.anyView().find(ViewType.Text.self)  // this return `nil`
```

And because it's an `AnyView`, we can't really read those `fixedSize()`, `hidden()` values.
Basically the following is now wrapped in an `AnyView`
```
content
   .fixedSize(horizontal: false, vertical: true)
    .hidden()
    .readHeight(callback) // used another view modifier here
```

I've also looked at this test and this isn't something we, as a host app, should do, right? https://github.com/nalexn/ViewInspector/blob/c70b1012453ca364cc846e21d041145664fe3460/Tests/ViewInspectorTests/SwiftUI/AnyViewTests.swift#L19

That'll require @testable import which makes those `internal` things available in the host project, which is strange to me.

**Comment by nalexn** [comment]
In this situation:

```swift
AnyView( // implicit
  content
    .fixedSize(horizontal: false, vertical: true)
    .hidden()
    .readHeight(callback) // used another view modifier here
)
```
considering you dug to the implicit AnyView, you can unwrap the `content` of the view modifier with `.viewModifierContent()` call. And from there `.fixedSize` and other modifiers should be accessible.
Alternatively, you can use `find(ViewType.ViewModifierContent.self)` earlier in the process, like so:

```swift
let content = try testView.inspect().find(ViewType.ViewModifierContent.self, skipFound: 1)
```

That should give you the second `content` from the background.
